src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1983,38 +1983,49 @@ public void RemoveProjectOutputPath_NoLock(SolutionChangeAccumulator solutionCha
 
         private void RefreshMetadataReferencesForFile(object sender, string fullFilePath)
         {
-            using (_gate.DisposableWait())
+            var asyncToken = _workspaceListener.BeginAsyncOperation(nameof(RefreshMetadataReferencesForFile));
+            _threadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
```

Comment:
we have a .CompletesAsyncToken extension so you don't need the try/finally

Reply:
oh wait.. is this one of those weird tasks?

Reply:
The try/finally was gotten rid of per @sharwell's suggestion to use just the using var token; in this case it's returning JoinableTasks so I'm not sure our existing helpers work, and as @sharwell pointed out we don't need it since RunWithShutdownBlockAsync still runs the lambda immediately.

Reply:
(GitHub should be showing this comment as outdated since this was already fixed, not sure but @CyrusNajmabadi broke it somehow.)